client/app.jsx

```diff
@@ -49,28 +66,48 @@ export default class App extends React.Component {
 
   renderContent() {
     const { path } = this.state.route;
-    const deck = this.state.route.params.get('deckId');
-    const name = this.state.route.params.get('deckName');
+    const { deckId, deckName, cardId, cards } = this.state;
     if (path === '') {
       return <Decks decks={this.state.decks} showModal={this.showModal} />;
     } else if (path === 'add-card') {
-      return <AddCard deckId={deck} deckName={name} />;
+      return <AddCard deckId={deckId} deckName={deckName} />;
     } else if (path === 'view-cards') {
-      return <ViewCards deckId={deck} deckName={name} cards={this.state.cards} />;
+      return <ViewCards deckId={deckId} deckName={deckName} cards={cards} />;
     } else if (path === 'edit-card') {
-      const card = this.state.route.params.get('cardId');
-      return <EditCard deckId={deck} deckName={name} cardId={card} />;
+      return (
+        <EditCard
+        deckId={deckId}
+        deckName={deckName}
+        cardId={cardId}
+        showModal={this.showModal}
+      />);
     } else {
       return <NotFound />;
     }
   }
 
+  renderModalForm() {
+    const { form, deckId, deckName, cardId } = this.state;
```

Comment:
I believe we can grab most if not all of this info from our route params.

Reply:
Same change made down here

![image](https://user-images.githubusercontent.com/109567499/201175038-e933f62e-6d15-4b31-a9a7-d5544eb03f4a.png)